app/components/chat.tsx

```
@@ -145,74 +147,65 @@ export function Chat({ repo, path, setPath, graph, chartRef, selectedPathId, isP
-    useEffect(() => {
-        setIsPath(isPathResponse)
     }, [isPathResponse])
 
-    const updatePreviousPath = (chart: cytoscape.Core, p: { nodes: any[], edges: any[] }) => {
+    const handelSetSelectedPath = (p: PathData) => {
         setSelectedPath(prev => {
             if (prev) {
-                if (isPathResponse && paths.some((path) => [...path.nodes, ...path.edges].every((e: any) => [...prev.nodes, ...prev.edges].some((el: any) => el.id === e.id)))) {
-                    chart.edges().forEach(e => {
-                        const id = e.id()
+                if (isPathResponse && paths.some((path) => [...path.nodes, ...path.links].every((e: any) => [...prev.nodes, ...prev.links].some((e: any) => e.id === e.id)))) {
```

Biome (1.9.4)

[error] Comparing to itself is potentially pointless.



(lint/suspicious/noSelfCompare)